### PR TITLE
[Fixes #4636 ] Docker install - build

### LINKED ISSUE #4636: Docker install - build
## Expected Behavior
To install Geonode using the docker install instructions

## Actual Behavior
On building the Django part, the build fails with:

Building django
Step 1/9 : FROM geonode/geonode:2.10
ERROR: Service 'django' failed to build: manifest for geonode/geonode:2.10 not found

## Steps to Reproduce the Problem

  1.  Install Docker for Windows
  2.  Clone Repository
  3.  Execute  docker-compose -f docker-compose.yml -f docker-compose.override.localhost.yml up --build in geonode folder

## Specifications

  - GeoNode version:
  - Installation method :  Geonode Docker
  - Platform:  Windows 10
  - Additional details:  Master Repository 

This is a bit of a bummer.  Was really looking forward in using Geonode...

Anyway, I will try just pulling the images, and not building.  Would that at least for now solve my problem?

What version is Master?

Pieter

### PR DESCRIPTION
**Pull request by gannebamm**
The dockerimage tags were wrong. They were updated by @francbartoli to follow GN-major.minor naming convention. This PR does change the dockerfiles to align with the new convention.

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [ x] Confirm you have read the [contribution guidelines](https://github.com/GeoNode/geonode/blob/master/CONTRIBUTING) 
- [ ] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)

I can not sign this CLA for my company without discussing it within the corporation (research institute - www.thuenen.de). I am contributing as single person currently.

- [ ] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

I will not target master branch, but 2.10.x, as discussed on gitter on [July 10, 2019 11:00 AM](https://gitter.im/GeoNode/general?at=5d25a93117cc7b05caa2f985)

The following are required only for core and extension modules (they are welcomed, but not required, for contrib modules):
- [x] There is a ticket in https://github.com/GeoNode/geonode/issues describing the issue/improvement/feature (a notable exemptions is, changes not visible to end users)
- [x] The issue connected to the PR must have Labels and Milestone assigned
- [x] PR for bug fixes and small new features are presented as a single commit
- [x] Commit message must be in the form "[Fixes #<issue_number>] Title of the Issue"
- [ ] New unit tests have been added covering the changes, unless there are explanation on why the tests are not necessary/implemented
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] This PR passes the QA checks: flake8 geonode
- [ ] Commits changing the **settings**, **UI**, **existing user workflows**, or adding **new functionality**, need to include documentation updates

**Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.**

### CODE REVIEW

**Comment by francbartoli** [comment]
Thanks @gannebamm but the tag for 2.10.x images is still there on docker hub. I would change the tag into 2.10.x until we are quite much more confident that there aren't problems with the installation. After all, GN-2.10 is just a mirror, nothing more. Also, it is convenient for me that the GN-2.10 docker hub tag will be present only in the GitHub tag 2.10 and 2.10.x from docker hub continues to be aligned with the branch 2.10.x.

**Comment by gannebamm** [comment]
@francbartoli ok. Thats different from what was discussed on gitter. So you would like to have the `2.10.x` tag instead `GN-2.10`?

If the `GN-2.10` will later be the stable release, shouldn´t we just use it as tag and if there are errors fix them? Switching to `2.10.x` now will lead to later switch to `GN-2.10` if the release process is finished?

**Comment by francbartoli** [comment]
@afabiani sorry I'm a little bit confused here:

- why is there the tag `backport 2.10.x`? This PR has been already done against that branch
- Also this merge mixes things up with the automation on docker hub because:

1. images with `GN-2.10` tag are automatically built from a GitHub tag namely for each repository `<name>-docker `

2. as soon as the branch `2.10.x` for `geonode-docker` receive new commits a new image is created while this won't happen unless the GitHub tag `2.10` is automatically created (we haven't this kind of automation yet) 

Long story short, it would be better IMHO to keep the tag for the image `geonode/geonode` at version `2.10.x` (aligned with the branch) and to go with a trigger at any time we want to release manually an updated geonode `2.10` tag on GitHub by using the tag `GN-2.10` from docker hub.

**Comment by afabiani** [comment]
@francbartoli should we revert?

**Comment by afabiani** [comment]
@francbartoli @gannebamm btw, PRs must be done against "master" branch first. I haven't noticed that. I guess we need to revert this.

**Comment by francbartoli** [comment]
Yep @afabiani, I think it is better to revert

**Comment by afabiani** [comment]
@francbartoli there's something that I don't understand here.
Why are you saying that GN-2.10 is built every time? From the docker hub configuration, it looks like it is pointing to a GitHub tag, am I wrong?

image

**Comment by francbartoli** [comment]
Exactly, I'm fully saying the contrary since it is building against a GitHub tag. I guess, well I would expect better, that if the tag is updated on GitHub then the image with tag GN-2.10 would be built again. So in term of repository code, we have to make sure the GH tag is gone to be updated for a new artifact from Docker Hub

**Comment by afabiani** [comment]
A tag shouldn't be updated!

**Comment by francbartoli** [comment]
Btw only the docker-compose scripts were updated with the frozen version GN-2.10 for the images
